src/components/directory-content.jsx

```diff
@@ -293,93 +263,52 @@ const DirectoryContent = () => {
                         hasMetadata:
-                            childrenMetadata[event.rowData.elementUuid] !==
+                            childrenMetadata[event.data.elementUuid] !==
                             undefined,
                         specificMetadata:
-                            childrenMetadata[event.rowData.elementUuid]
+                            childrenMetadata[event.data.elementUuid]
                                 ?.specificMetadata,
-                        ...element,
+                        ...event.data,
                     });
-
-                    if (contextualMixPolicy === contextualMixPolicies.BIG) {
-                        // If some elements were already selected and the active element is not in them, we deselect the already selected elements.
-                        if (
-                            selectedUuids?.size &&
-                            element?.elementUuid &&
-                            !selectedUuids.has(element.elementUuid)
-                        ) {
-                            setSelectedUuids(new Set());
-                        }
-                    } else {
-                        // If some elements were already selected, we add the active element to the selected list if not already in it.
-                        if (
-                            selectedUuids?.size &&
-                            element?.elementUuid &&
-                            !selectedUuids.has(element.elementUuid)
-                        ) {
-                            let updatedSelectedUuids = new Set(selectedUuids);
-                            updatedSelectedUuids.add(element.elementUuid);
-                            setSelectedUuids(updatedSelectedUuids);
-                        }
-                    }
                 }
                 setMousePosition({
                     mouseX: event.event.clientX + constants.HORIZONTAL_SHIFT,
                     mouseY: event.event.clientY + constants.VERTICAL_SHIFT,
                 });
                 handleOpenContentMenu(event.event);
-            } else {
+            }
+        },
+        [childrenMetadata]
+    );
+
+    const onContextMenu = useCallback(
+        (event) => {
+            if (!isRowHovered()) {
+                if (selectedDirectory) {
+                    dispatch(setActiveDirectory(selectedDirectory.elementUuid));
+                }
                 setMousePosition({
                     mouseX: event.clientX + constants.HORIZONTAL_SHIFT,
                     mouseY: event.clientY + constants.VERTICAL_SHIFT,
                 });
                 handleOpenDirectoryMenu(event);
             }
         },
-        [
-            currentChildren,
-            dispatch,
-            selectedDirectory,
-            selectedUuids,
-            contextualMixPolicies,
-            contextualMixPolicy,
-            childrenMetadata,
-        ]
+        [dispatch, selectedDirectory]
     );
```

Comment:
is it to handle right click on headers ? 
Or is it to prevent multiple contextual menus to open when right clicking on aggrid ?
It sounds complicated with the "isRowHovered", and it does not seem to work properly (check last comment about bugs)

Reply:
It handles context menu event on the container wrapping the grid
To prevent the bug I now handle every context menu event through this one instead of using AG grid built in cell context menu event. 
There's no visual glitches anymore and the only drawback is when you try to open a row context menu while another one is already open it will open the general context menu